CONTRIBUTING.md

```diff
@@ -59,18 +59,25 @@ The `metadata.json` file contains metadata of the module including `name`, `desc
   // The name of the module (10 - 60 characters).
   "name": "Sample module",
   // The description of the module (10 - 1000 characters).
-  "description": "Sample module description",
+  "summary": "Sample module description. This should be a short description of the functionality of the module. A more detailed description can be included in the README.md file.",
   // The owner of the module. Must be a GitHub username or a team under the Azure organization
   "owner": "sampleusername"
 }
 
 ```
 
+NOTE: The metadata.json is in the process of being deprecated, in favor of metadata in the `main.bicep` file (see below). For the moment, the `metadata.json` file is still REQUIRED. It is RECOMMENDED but not currently required that you also add the information in `summary.json` to `bicep.main` as follows: Copy `"name"`, `"summary"` and `"owner"` from `metadata.json` into `metadata` statements in `main.bicep` with the names `'name'`, `'description'` and `'owner'`. (Note the change from `summary` to `description`.)
+
 #### `main.bicep`
 
 The `main.bicep` file is the public interface of the module. When authoring `main.bicep`, make sure to provide a description for each parameter and output. You are free to create other Bicep files inside the module folder and reference them as local modules in `main.bicep` if needed. You may also reference other registry modules to help build your module. If you do so, make sure to add them as external references with specific version numbers. You should not reference other registry modules through local file path, since they may get updated overtime. The `main.bicep` most follow various static code analysis checks, such as including descriptions on every parameter and output. This will be used to automatically generate the `README.md`.
 
 ```bicep
+metadata name = 'Sample module'
+metadata description = '''Sample module description. This should be a short description of the functionality of the module.
+A more detailed description can be included in the README.md file.'''
```

Comment:
Just to make sure, we'll add metadata `details` to `main.bicep` once the `brm` tool is updated, right?

Reply:
No, the decision was the opposite. From Teams (luckily I wrote it down, too confusing to remember):
> Decision on "summary/description" vs "description/details".  We decided to go ahead and switch terminology from the current summary/description to "description/details" (as we had decided late last week).  This means the following:
>  
> 1) The "summary" property from metadata.json, which is a short documentation string will be switched to using "metadata description" in the bicep file.
> 
> 2) "Description" section in README.md will be renamed to "Details".

Work tracked here: https://github.com/Azure/bicep/issues/10810

I can't change Description in README.md until I update the tool.